src/tetris/brick/brick.ts

```diff
@@ -71,9 +80,52 @@ export default class Brick {
 	//region private members
 	private _position: Vector2;
 	private _blocks: Block[];
-	private readonly _blockAssetId: string;
+	private _field: Field;
+	private _stuck: boolean = false;
 	//endregion
 
 	//region private methods
+	private _tryToMove(move: Vector2, stuckIfFails: boolean = false): void {
+		if (this._isMovePossible(move)) {
+			this.position.add(move);		
+		} else if (stuckIfFails) {
+			this._stuck = true;
+		}
+	}
+	
+	private _isMovePossible(move: Vector2): boolean {
+		let possible = true;
+		this._blocks.forEach(b => {
+			const targetPosition = b.position.clone().add(move).add(this._position);
+			if (targetPosition.x < 0 || targetPosition.y < 0 || targetPosition.x >= this._field.width || targetPosition.y >= this._field.height || this._field.state[targetPosition.x][targetPosition.y]) {				
+				possible = false;
+			}
+		})
+		return possible;
+	}
+
+	private _tryToRotate(clockwise: boolean): void {
+		const rotatedBlocks = this._blocks.map(originalBlock => {
+			let rotatedBlock = originalBlock.clone();
+			rotatedBlock.position = clockwise ? new Vector2(-rotatedBlock.position.y, rotatedBlock.position.x) : new Vector2(rotatedBlock.position.y, -rotatedBlock.position.x); 
```

Comment:
Please do not use inline if. They decrease understandability and provoke errors in many case